Request for a standard CODEOWNERS file.

ReilleyMilne opened the pull request:
# Request for a standard CODEOWNERS file.

CODEOWNER changes:

- Sorted both the Management Plane and Services categories, every other category is left alone. Everything is now sorted by service label while maintaining path priority (Also prioritizes wildcards).
- /sdk/template/ is above /sdk/ so it will never be called. This existed before I sorted the file.
- In the code generation summary, all the PR labels are as such #PRLabel, when correct formatting should be # PRLabel. I'm not sure if this would make it so that they're not being used properly but just a note.
- Below is some of the changes that were found to be different (All of these are just reformatting), I have a change log that compares all the removed + added content to make sure that no data is being erased.

# Git Diff Summary

Generated: 2025-08-06 11:00:41

## Statistics
- Lines removed: 194
- Lines added: 199  
- Net change: 5
- Content preserved (moved): 136
- Content lost: 10
- New content: 14

## Validation Results
- Total path patterns: 155
- Duplicate paths: 0
- Validation issues: 0
**No validation issues found** - File appears properly organized

## Removed Content (10 lines)
- `# ######## Services ########`
- `# ServiceOwners:                     	                             @bowgong @mengaims`
- `# ServiceOwners:                                                   @@seanmcc-msft`
- `#ServiceLabel: %Remote Rendering`
- `#ServiceOwners:                                                    @FlorianBorn71 @MichaelZp0 @ChristopherManthei`
- `# ######## Management Plane ########`
- `# PRLael: %Container Orchestrator Runtime`
- `# ServiceLabel: %Durable Task Scheduler  %Mgmt`
- `# ServiceLabel:  %Network - Front Door %Mgmt`
- `# ServiceLabel: %%Migration Assessment %Mgmt`

## Added Content (14 lines)
- `####################`
- `# Client Libraries`
- `####################`
- `# ServiceOwners:                                                   @bowgong @mengaims`
- `# ServiceOwners:                                                   @seanmcc-msft`
- `# ServiceLabel: %Remote Rendering`
- `# ServiceOwners:                                                   @FlorianBorn71 @MichaelZp0 @ChristopherManthei`
- `####################`
- `# Management Libraries`
- `####################`
- `# PRLabel: %Container Orchestrator Runtime`
- `# ServiceLabel: %Durable Task Scheduler %Mgmt`
- `# ServiceLabel: %Migration Assessment %Mgmt`
- `# ServiceLabel: %Network - Front Door %Mgmt`

ReilleyMilne committed:
Update CODEOWNERS

ReilleyMilne committed:
Update CODEOWNERS

jsquire commented on .github/CODEOWNERS:
```diff
@@ -78,7 +78,9 @@
 
-# ######## Services ########
+####################
+# Client SDKs
```
```suggestion
# Client Libraries
```

nit:  The term "Azure SDK" is all of the things.   The individual bits are "libraries" or "packages"

ReilleyMilne committed:
Update .github/CODEOWNERS

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

jsquire commented on .github/CODEOWNERS:
```diff
@@ -930,7 +932,9 @@
 
-# ######## Management Plane ########
+####################
+# Management Plane SDKs
```
```suggestion
# Management Libraries
```

jsquire replied:
We want to normalize here too?

ReilleyMilne committed:
Update .github/CODEOWNERS

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

ReilleyMilne committed:
Update CODEOWNERS

Sorted the CODEOWNERS file. It's currently being sorted by path first and then service/pr label. This does push all the services not connected to a path to the bottom of their category. Will need some review here!

jsquire commented on .github/CODEOWNERS:
```diff
@@ -98,99 +97,36 @@
 # ServiceLabel: %AI Projects
 # ServiceOwners:                                                   @dargilco @jhakulin @nick863
 
-# ServiceLabel: %AKS
```
Why are we doing anything beyond changing the headings?    Please explain why I'm seeing bit swaths of removes and adds.

ReilleyMilne committed:
Update CODEOWNERS

CODEOWNER changes:
Sorted both the Management Plane and Services categories, every other category is left alone. Everything is now sorted by service label while maintaining path priority (Also prioritizes wildcards).
/sdk/template/ is above /sdk/ so it will never be called. This existed before I sorted.
In the code generation summary, all the PR labels are as such #PRLabel, when correct formatting should be # PRLabel. I'm not sure if this would make it so that they're not being used properly but just an note.
Below is some of the changes that were found to be different (All of these are just reformatting), I have a change log that compares all the removed + added content to make sure that no data is being erased.

# Git Diff Summary

Generated: 2025-08-06 11:00:41

## Statistics
- Lines removed: 194
- Lines added: 199  
- Net change: 5
- Content preserved (moved): 136
- Content lost: 10
- New content: 14

## Validation Results
- Total path patterns: 155
- Duplicate paths: 0
- Validation issues: 0
**No validation issues found** - File appears properly organized

## Removed Content (10 lines)
- `# ######## Services ########`
- `# ServiceOwners:                     	                             @bowgong @mengaims`
- `# ServiceOwners:                                                   @@seanmcc-msft`
- `#ServiceLabel: %Remote Rendering`
- `#ServiceOwners:                                                    @FlorianBorn71 @MichaelZp0 @ChristopherManthei`
- `# ######## Management Plane ########`
- `# PRLael: %Container Orchestrator Runtime`
- `# ServiceLabel: %Durable Task Scheduler  %Mgmt`
- `# ServiceLabel:  %Network - Front Door %Mgmt`
- `# ServiceLabel: %%Migration Assessment %Mgmt`

## Added Content (14 lines)
- `####################`
- `# Client Libraries`
- `####################`
- `# ServiceOwners:                                                   @bowgong @mengaims`
- `# ServiceOwners:                                                   @seanmcc-msft`
- `# ServiceLabel: %Remote Rendering`
- `# ServiceOwners:                                                   @FlorianBorn71 @MichaelZp0 @ChristopherManthei`
- `####################`
- `# Management Libraries`
- `####################`
- `# PRLabel: %Container Orchestrator Runtime`
- `# ServiceLabel: %Durable Task Scheduler %Mgmt`
- `# ServiceLabel: %Migration Assessment %Mgmt`
- `# ServiceLabel: %Network - Front Door %Mgmt`

ReilleyMilne committed:
Update CODEOWNERS

Added the new changes made by the recent PR by Jesse.

jsquire reviewed (approved):
Nice!  That reorder looks more like what I'd expect (and fixes some really old entries)

ReilleyMilne committed:
Merge branch 'main' into codeowners_category_update

ReilleyMilne commented:
@microsoft-github-policy-service agree company="Microsoft"

ReilleyMilne committed:
Merge branch 'main' into codeowners_category_update

ReilleyMilne committed:
Update CODEOWNERS

jsquire commented on .github/CODEOWNERS:
```diff
@@ -104,13 +109,22 @@
+# ServiceLabel: %App Services
+# ServiceOwners:                                                   @antcp @AzureAppServiceCLI
+
+# ServiceLabel: %ARM
```
If you're feeling like taking that on, it would be greatly appreciated.   Non-blocking for this moving forward, if you'd prefer to leave it be.

ReilleyMilne committed:
Update CODEOWNERS

Added the ARM services to the Management Libraries category.

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
 
-# PRLabel: %Arize AI
-/sdk/arizeaiobservabilityeval/Azure.ResourceManager.*/             @aggarwalsw
+# ServiceLabel: %ARM
```
```suggestion
# ServiceLabel: %ARM %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
 
-# ServiceLabel: %Arize AI %Mgmt
-# ServiceOwners:                                                   @aggarwalsw
+# ServiceLabel: %ARM - Core
```
```suggestion
# ServiceLabel: %ARM - Core %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
+# ServiceLabel: %ARM - Core
+# ServiceOwners:                                                   @armleads-azure
+
+# ServiceLabel: %ARM - Managed Applications
```
```suggestion
# ServiceLabel: %ARM - Managed Applications %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
+# ServiceLabel: %ARM - Managed Applications
+# ServiceOwners:                                                   @armleads-azure
+
+# ServiceLabel: %ARM - RBAC
```
```suggestion
# ServiceLabel: %ARM - RBAC %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
+# ServiceLabel: %ARM - RBAC
+# ServiceOwners:                                                   @armleads-azure
+
+# ServiceLabel: %ARM - Service Catalog
```
```suggestion
# ServiceLabel: %ARM - Service Catalog %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
+# ServiceLabel: %ARM - Service Catalog
+# ServiceOwners:                                                   @armleads-azure
+
+# ServiceLabel: %ARM - Tags
```
```suggestion
# ServiceLabel: %ARM - Tags %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
+# ServiceLabel: %ARM - Tags
+# ServiceOwners:                                                   @armleads-azure
+
+# ServiceLabel: %ARM - Templates
```
```suggestion
# ServiceLabel: %ARM - Templates %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -982,7 +986,7 @@
+# PRLabel: %Container Orchestrator Runtime
 /sdk/containerorchestratorruntime/Azure.ResourceManager.*/         @ArthurMa1978
 
 # ServiceLabel: %Container Orchestrator Runtime
```
```suggestion
# ServiceLabel: %Container Orchestrator Runtime  %Mgmt
```

jsquire commented on .github/CODEOWNERS:
```diff
@@ -955,17 +938,38 @@
 # ServiceLabel: %API Center %Mgmt
 # ServiceOwners:                                                   @alexyaang
 
+# PRLabel: %Arize AI
```
```suggestion
# PRLabel: %Arize AI %Mgmt
```

jsquire reviewed (commented):
If you can just pull in the suggestions to align with the management area format, I think we're good.

ReilleyMilne committed:
Apply suggestions from code review

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>
